core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java

```diff
@@ -152,4 +181,84 @@ private void saveDlpPositiveItem(Node node) throws RepositoryException {
     dlpPositiveItemEntity.setKeywords(dlpKeywords);
     dlpPositiveItemService.addDlpPositiveItem(dlpPositiveItemEntity);
   }
+
+  private void removeRestorePathInfo(Session session, String restorePath) throws Exception {
+    Node sameNameNode = ((Node) session.getItem(restorePath));
+    Node parent = sameNameNode.getParent();
+    String name = sameNameNode.getName();
+    NodeIterator nodeIter = parent.getNodes(name);
+    while (nodeIter.hasNext()) {
+      Node node = nodeIter.nextNode();
+      if (node.isNodeType(EXO_RESTORE_LOCATION))
+        node.removeMixin(EXO_RESTORE_LOCATION);
+    }
+  }
+
+  private void addRestorePathInfo(String nodeName, String restorePath, String nodeWorkspace) throws Exception {
+    NodeIterator nodes = this.getSecurityHomeNode().getNodes(nodeName);
+    Node node = null;
+    while (nodes.hasNext()) {
+      Node currentNode = nodes.nextNode();
+      if (node == null) {
+        node = currentNode;
+      } else {
+        if (node.getIndex() < currentNode.getIndex()) {
+          node = currentNode;
+        }
+      }
+    }
+    if (node != null) {
+      node.addMixin(EXO_RESTORE_LOCATION);
+      node.setProperty(RESTORE_PATH, restorePath);
+      node.setProperty(RESTORE_WORKSPACE, nodeWorkspace);
+      node.save();
+    }
+  }
+
+  private Node getSecurityHomeNode() {
+    try {
+      Session session = WCMCoreUtils.getSystemSessionProvider()
+                                    .getSession(COLLABORATION_WS,
+                                                repositoryService.getCurrentRepository());
+      return (Node) session.getItem("/" + DLP_SECURITY_FOLDER);
+    } catch (Exception e) {
+      return null;
+    }
+
+  }
+
+  private String fixRestorePath(String path) {
```

Comment:
Can you explain what is this for ?

Reply:
This method is used to fix the path if it contains '[ 'or ']'.

Reply:
In which case the path contains [ ?
And why should we fix that ?
What is not working ?

Reply:
cc @ahmadmido